File: front/src/app/models/Article.ts
import { User } from './User';

export interface Article {
  id: number;
  topicId: number;
  title: string;
  createdAt: string;
  user: User;
  content: string;
}

export interface ArticleRequest {
  topicId: number;
  title: string;
  content: string;
  userId: number;
}

File: back/src/main/java/com/openclassrooms/mddapi/models/Article.java
package com.openclassrooms.mddapi.models;

import javax.persistence.*;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import lombok.*;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "articles")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Article {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "topic_id", nullable = false)
    private Topic topic;

    private String title;

    @Column(columnDefinition = "TEXT")
    private String content;

    private LocalDateTime createdAt;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @OneToMany(mappedBy = "article", cascade = CascadeType.ALL)
    @JsonManagedReference
    private List<Comment> comments = new ArrayList<>();
}
